drivers/builtin/src/aes.c

```diff
@@ -480,6 +480,36 @@ void mbedtls_aes_free(mbedtls_aes_context *ctx)
     mbedtls_platform_zeroize(ctx, sizeof(mbedtls_aes_context));
 }
 
+void mbedtls_aes_get_implementation(mbedtls_aes_implementation **aes_imp_out)
+{
+    int aes_imp = MBEDTLS_AES_IMP_UNKNOWN;
+#if defined(MBEDTLS_AESNI_HAVE_CODE)
+    if (mbedtls_aesni_has_support(MBEDTLS_AESNI_AES)) {
+#if MBEDTLS_AESNI_HAVE_CODE == 1
+        aes_imp = MBEDTLS_AES_IMP_AESNI_ASM;
+#elif MBEDTLS_AESNI_HAVE_CODE == 2
+        aes_imp = MBEDTLS_AES_IMP_AESNI_INTRINSICS;
+#endif /* MBEDTLS_AESNI_HAVE_CODE == 1 || 2 */
+        goto exit;
+    }
+#endif /* MBEDTLS_AESNI_HAVE_CODE */
+
+#if defined(MBEDTLS_AESCE_HAVE_CODE)
+    if (MBEDTLS_AESCE_HAS_SUPPORT()) {
+        aes_imp = MBEDTLS_AES_IMP_AESCE;
+        goto exit;
+    }
+#endif
+
+#if !defined(MBEDTLS_AES_USE_HARDWARE_ONLY)
+    aes_imp = MBEDTLS_AES_IMP_SOFTWARE;
+    goto exit;
+#endif
+
+exit:
+    *aes_imp_out = &aes_imp;
```

Comment:
This sets the out parameter to point to `aes_imp`, which is a stack-allocated variable that will disappear when the function returns, so this will cause undefined behaviour when the caller tries to access the out parameter.

This would also be fixed by returning the value `aes_imp` directly.